guides/concepts/index.md

```diff
@@ -1 +1,100 @@
+This section is a crash course in Cardstack concepts. It includes vocabulary that you'll find throughout the other topics, and it draws some parallels to other frameworks and JavaScript in general.
+
+
+## Essential Vocabulary
+
+### Cards
+
+Cards are reusable units of functionality - like a blog article, a text editor, a payment system, and more.
+A developer would use many cards, and cards inside of cards, to create the end user experience.
+Each card can have its own data schema, which flows from the front end to the data layer. When someone is looking at a card, it is reflected in the browser URL.
+The information contained in that URL informs communication with the back end servers.
+
+### The Cardstack mono-repo
+
+The [Cardstack mono-repo](https://github.com/cardstack/cardstack) is a collection of the various elements that could be used to create a Cardstack environment.
+It contains some commonly-used Cards, the Cardstack Hub that connects data sources, and the code that ties everything together behind the scenes.
+
+### Plugins and Data Adapters
+
+A plugin, or data adapter, is made up of four functions needed to connect with a new data source.
+For example, someone could write or use plugins to connect to GitHub, Firebase, and a blockchain source.
+That same plugin code could be used by anyone running a Cardstack environment who wants to connect to the same data source.
+
+### The Hub
+
+The Cardstack Hub is where data from many sources come together.
+Developers can activate any number of plugins provided by Cardstack or insert their own.
+The Cardstack Hub does all the heavy lifting that takes the data retrieved by plugins and makes it available to the front end.
+To connect to a new data source, a developer only has to write a plugin, not modify the Hub. The Hub's code is contained within the Cardstack monorepo.
+
+### The Cardstack Environment
+
+It’s helpful to think of Cardstack “environments” rather than "apps." 
+Although Cardstack is built for the web and not native, in some ways its mental model is more similar to an operating system than a web app.
+For example, say you make a Word doc, and then share it with someone else, who then makes revisions and sends it back to you via email.
+That’s analogous to a developer composing a Card, making it available for others to use, and accepting revisions. Someone could build an app _using_ Cardstack, but Cardstack itself is not an app.
+
+### Schema
+
+Schema is a general term to describe the shape that data takes. What attributes come back from an API request? When developing with Cardstack, you may define the schema that influences how data is stored, its relationships, and what it is named when it is used on the front end.
+
+### Indexers, Searchers, and Writers
```

Comment:
Should this be a subsection of the `The Hub`? I think it's important to mention that indexers, searchers, and writers are similar to CRUD in a REST server, except that that the data sources can be from anywhere. Maybe we can have a one sentence description of each thing, i.e.

Indexer - an asynchronous process that fetches data from external sources, does some JSON preprocessing, and add the data into an index to allow for quick access when the front end requests for it.

Searcher - when querying/fetching data from the front end, the searcher will access the data from the index.

Writer - when the frontend sends a POST/PATCH request (want to write back to the data source), the writer handles that.

I'm not sure if these one lines are accurate, but having something like this could really help readers wrap their heads around what these things are.

Reply:
I added this section into the Plugins explanation